src/components/mdx_wrapper.tsx

```diff
@@ -33,6 +33,7 @@ const components = {
   table: TableWrapper,
   TextInput: TextInputWrapper,
   USStateSelect: USStateSelectWrapper,
+  Date: DateStr,
```

Comment:
To avoid confusion with the builtin JS Date object, I suggest changing the name of the shortcode to `DateStr`.  I'd change the file name to match.